config/env_spec.go

```diff
@@ -602,3 +611,101 @@ type CorsPolicy struct {
 func (c CorsPolicy) IsEmpty() bool {
 	return len(c.AllowOrigins) == 0 && len(c.AllowOriginsRegexes) == 0
 }
+
+// TargetAuthentication configures how to authenticate the request to the backend.
+type TargetAuthentication struct {
+	// The time between two adjacent token refreshments.
+	RefreshInterval time.Duration `yaml:"refresh_interval,omitempty" mapstructure:"refresh_interval,omitempty"`
+
+	// GoogleOAuth configures how to authenticate the request to the backend with Google OAuth tokens.
+	GoogleOAuth GoogleOAuth `yaml:"google_oauth,omitempty" mapstructure:"google_oauth,omitempty"`
+
+	// TokenSource is used to fetch OAuth token.
+	// This is not part of the yaml config and will be generated internally.
+	TokenSource oauth2.TokenSource `yaml:"-" mapstructure:"-"`
```

Comment:
What is the purpose of this?

Reply:
This is to generate the oauth token. I am implementing it in a different pkg. If we have other providers, we can use this same interface. The plan is to initiate this on loading the config so that it is easier to access.

Reply:
That would require maps of maps to track operation level token source. I think this is very similar to the jwtAuthentications, which was put in the config for the relative easiness to track. I can make this internal and add a accessor method if that makes more sense.

Reply:
I see. It seems fine. Without the implementation, it's just hard to see how it's being used.

Reply:
Yeah before merging this, I will show you a draft PR in -golib so it's clearer how token source is configured and consumed.